### ISSUE TITLE
How to replicate text-to-video  results on MSR-VTT dataset

### ISSUE THREAD

**Issue by Alen-T** [Jiamian-Wang/T-MASS-text-video-retrieval#5]
Dear authors,

First of all, thank you very much for your outstanding work and for sharing the code. Using the same device（A6000 GPU） and environment settings you shared, we obtained the results 'R@1: 46.4 (window: 46.4)', 'R@5: 74.9 (window: 74.9)', 'R@10: 84.5 (window: 84.5)', 'MedR: 2.0 (window: 2.0)', 'MeanR: 13.128 (window: 13.128)'. We would like to know how to replicate the results mentioned in the paper. Thanks！

**Comment by Jiamian-Wang** [comment]
Could you share the specific training setting?

**Comment by Alen-T** [comment]
config={'dataset_name': 'MSRVTT', 'videos_dir': '{video_dir}', 'msrvtt_train_file': '9k', 'num_frames': 12, 'video_sample_type': 'uniform', 'input_res': 224, 'exp_name': 'MSR-VTT-9k', 'model_path': './outputs/MSR-VTT-9k/2024_08_04_12_59_49', 'output_dir': './outputs', 'save_every': 1, 'log_step': 10, 'evals_per_epoch': 10, 'load_epoch': None, 'eval_window_size': 5, 'metric': 't2v', 'arch': 'clip_stochastic', 'huggingface': True, 'clip_arch': 'ViT-B/32', 'embed_dim': 512, 'loss': 'clip', 'clip_lr': 1e-06, 'noclip_lr': 3e-05, 'batch_size': 32, 'num_epochs': 5, 'weight_decay': 0.2, 'warmup_proportion': 0.1, 'pooling_type': None, 'k': -1, 'attention_temperature': 0.01, 'num_mha_heads': 1, 'transformer_dropout': 0.3, 'num_workers': 8, 'seed': 24, 'no_tensorboard': False, 'tb_log_dir': 'logs/MSR-VTT-9k', 'stochasic_trials': 20, 'datetime': 2024_08_04_12_59_49, 'gpu': None, 'support_loss_weight': 0.8, 'batch_size_split': None, 'chunk_size': 128, 'noloss_record': False, 'save_memory_mode': False, 'raw_video': False, 'skip_eval': False, 'DSL': False, 'stochastic_prior': 'uniform01', 'stochastic_prior_std': 1.0}

**Comment by Alen-T** [comment]
The parameters are consistent with those in your paper, with no special settings applied.

**Comment by zef1611** [comment]
> config={'dataset_name': 'MSRVTT', 'videos_dir': '{video_dir}', 'msrvtt_train_file': '9k', 'num_frames': 12, 'video_sample_type': 'uniform', 'input_res': 224, 'exp_name': 'MSR-VTT-9k', 'model_path': './outputs/MSR-VTT-9k/2024_08_04_12_59_49', 'output_dir': './outputs', 'save_every': 1, 'log_step': 10, 'evals_per_epoch': 10, 'load_epoch': None, 'eval_window_size': 5, 'metric': 't2v', 'arch': 'clip_stochastic', 'huggingface': True, 'clip_arch': 'ViT-B/32', 'embed_dim': 512, 'loss': 'clip', 'clip_lr': 1e-06, 'noclip_lr': 3e-05, 'batch_size': 32, 'num_epochs': 5, 'weight_decay': 0.2, 'warmup_proportion': 0.1, 'pooling_type': None, 'k': -1, 'attention_temperature': 0.01, 'num_mha_heads': 1, 'transformer_dropout': 0.3, 'num_workers': 8, 'seed': 24, 'no_tensorboard': False, 'tb_log_dir': 'logs/MSR-VTT-9k', 'stochasic_trials': 20, 'datetime': 2024_08_04_12_59_49, 'gpu': None, 'support_loss_weight': 0.8, 'batch_size_split': None, 'chunk_size': 128, 'noloss_record': False, 'save_memory_mode': False, 'raw_video': False, 'skip_eval': False, 'DSL': False, 'stochastic_prior': 'uniform01', 'stochastic_prior_std': 1.0}

Hi can you try to train with DSL argument True? I am really curious of the performance with DSL on. Thank you a lot
